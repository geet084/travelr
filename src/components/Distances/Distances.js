import React from 'react';
import Collapsible from 'react-collapsible';
import Counter from '../Counter/Counter';

export const Distances = (props) => {
  const { timeInSeconds, earthSpin, earthOrbit, solarSystemOrbit, galaxyMovement, totalMovement } = props;

  return (
    <Collapsible trigger='Show More'>
      <div className='stats text-bg'>
        <h4>HOW FAST YOU ARE MOVING RIGHT NOW</h4>
        <h6>distance from new york to san francisco: <span>2,569 miles</span></h6>
        <h6>distance from san francisco to paris: <span>5,560 miles</span> </h6>
        <h6>distance from new york to australia: <span>10,512 miles</span></h6>
        <h6>circumference of the earth: <span>24,901 miles</span></h6>
        <h6>distance to the moon: <span>238,900 miles</span></h6>
      </div>
      <section className="details">
        <div className="title text-bg">
          <p>elapsed time since you arrived on this page in seconds</p>
          <p>Distance covered in miles due to Earth spinning</p>
          <p>Distance covered in miles due to the Earth orbiting the Sun</p>
          <p>Distance covered in miles due to the Solar system orbiting in the galaxy</p>
          <p>Distance covered in miles due to the Milky Way moving towards Andromeda</p>
          <p>Your total distance covered in that time</p>
        </div>
        <div className="counter-info text-bg">
          <p><Counter start={timeInSeconds} end={timeInSeconds + 1000000} decimal={1} /></p>
          <p><Counter start={earthSpin} end={earthSpin + 280000} decimal={1} /></p>
          <p><Counter start={earthOrbit} end={earthOrbit + 18500000} /></p>
          <p><Counter start={solarSystemOrbit} end={solarSystemOrbit + 124450000} /></p>
          <p><Counter start={galaxyMovement} end={galaxyMovement + 70000000} /></p>
          <p><Counter start={totalMovement} end={totalMovement + 213230000} /></p>
        </div>
      </section>
    </Collapsible>
  )
}

export default Distances;