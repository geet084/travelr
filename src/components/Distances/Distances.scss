.text-bg > * {
  background-color: rgba(0,0,0,0.7);
  box-shadow: 0px 0px 5px 5px rgba(0,0,0,0.7);
}

.stats {
  width: 300px;
  margin: 0 auto;
  h4, h6 {
    padding: 2px;
    margin: 0;
  }
}

.details {
 display: flex;
 justify-content: center;
}

.title {
  text-align: right;
  margin: 10px 5px 10px 10px;
  height: 250px;
}

.counter-info {
  text-align: left;
  margin: 10px 10px 10px 5px;
  height: 250px;
  min-width: 100px;
}