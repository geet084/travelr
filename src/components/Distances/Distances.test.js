import React from 'react';
import { shallow } from 'enzyme';
import { Distances } from './Distances';

describe('Distances', () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      counter: jest.fn(),
      timeInSeconds: .01,
      earthSpin: (.01 * .28),
      earthOrbit: (.01 * 18.5),
      solarSystemOrbit: (.01 * 124.45),
      galaxyMovement: (.01 * 70),
      totalMovement: (.01 * 213.23),
    }
    wrapper = shallow(<Distances {...mockProps} />)
  });

  describe('initial state', () => {
    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});