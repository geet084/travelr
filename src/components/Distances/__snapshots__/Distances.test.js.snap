// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Distances initial state should match snapshot 1`] = `
<Collapsible
  className=""
  classParentString="Collapsible"
  contentContainerTagName="div"
  contentInnerClassName=""
  contentOuterClassName=""
  easing="linear"
  lazyRender={false}
  onClose={[Function]}
  onClosing={[Function]}
  onOpen={[Function]}
  onOpening={[Function]}
  open={false}
  openedClassName=""
  overflowWhenOpen="hidden"
  tabIndex={null}
  transitionCloseTime={null}
  transitionTime={400}
  trigger="Show More"
  triggerClassName=""
  triggerDisabled={false}
  triggerOpenedClassName=""
  triggerSibling={null}
  triggerStyle={null}
  triggerTagName="span"
>
  <div
    className="stats text-bg"
  >
    <h4>
      HOW FAST YOU ARE MOVING RIGHT NOW
    </h4>
    <h6>
      distance from new york to san francisco: 
      <span>
        2,569 miles
      </span>
    </h6>
    <h6>
      distance from san francisco to paris: 
      <span>
        5,560 miles
      </span>
       
    </h6>
    <h6>
      distance from new york to australia: 
      <span>
        10,512 miles
      </span>
    </h6>
    <h6>
      circumference of the earth: 
      <span>
        24,901 miles
      </span>
    </h6>
    <h6>
      distance to the moon: 
      <span>
        238,900 miles
      </span>
    </h6>
  </div>
  <section
    className="details"
  >
    <div
      className="title text-bg"
    >
      <p>
        elapsed time since you arrived on this page in seconds
      </p>
      <p>
        Distance covered in miles due to Earth spinning
      </p>
      <p>
        Distance covered in miles due to the Earth orbiting the Sun
      </p>
      <p>
        Distance covered in miles due to the Solar system orbiting in the galaxy
      </p>
      <p>
        Distance covered in miles due to the Milky Way moving towards Andromeda
      </p>
      <p>
        Your total distance covered in that time
      </p>
    </div>
    <div
      className="counter-info text-bg"
    >
      <p>
        <Counter
          decimal={1}
          end={1000000.01}
          start={0.01}
        />
      </p>
      <p>
        <Counter
          decimal={1}
          end={280000.0028}
          start={0.0028000000000000004}
        />
      </p>
      <p>
        <Counter
          end={18500000.185}
          start={0.185}
        />
      </p>
      <p>
        <Counter
          end={124450001.2445}
          start={1.2445000000000002}
        />
      </p>
      <p>
        <Counter
          end={70000000.7}
          start={0.7000000000000001}
        />
      </p>
      <p>
        <Counter
          end={213230002.1323}
          start={2.1323}
        />
      </p>
    </div>
  </section>
</Collapsible>
`;
